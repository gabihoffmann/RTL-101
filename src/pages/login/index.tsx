import { Button } from "../../components/button";

export function LoginPage() {
  return (
    <div>
      <h1>Login Page</h1>
      <p>Click button bellow to login</p>
      <Button label="Login" />
    </div>
  );
}
