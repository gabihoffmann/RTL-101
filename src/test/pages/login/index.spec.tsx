import { describe, expect, it } from "vitest";

import { render, screen } from "@testing-library/react";
import { LoginPage } from "../../../pages/login";

describe("Login page", () => {
  it("Should render the login page correctly", () => {
    const { container } = render(<LoginPage />);

    const button = screen.getByRole("button", { name: /login/i });

    expect(container).toMatchSnapshot();
    expect(button).toBeInTheDocument();
  });
});
