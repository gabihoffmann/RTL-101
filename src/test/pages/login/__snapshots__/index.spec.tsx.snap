// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should  test login page successfully > Should render the login page correctly 1`] = `
<div>
  <div>
    <h1>
      Login Page
    </h1>
    <p>
      Click button bellow to login
    </p>
    <button>
      <strong>
        Login
      </strong>
    </button>
  </div>
</div>
`;
